packages/itwin/grouping-mapping-widget/src/widget/components/viewerUtils.tsx

```diff
@@ -138,7 +138,7 @@ export const overrideElements = (
     vp,
     ColorDef.fromString(color),
     overrideType,
-    true,
+    false,
```

Comment:
Do we somehow ensure that something clears the EmphasizeElements before calling `overrideElements` when we want to change the emphasis? Like from emphasizing one group of elements and then switching to a completely different group of elements?

Reply:
There is nothing guaranteeing that there is no overridden/emphasized elements. Just a blanket call of `clearEmphasizedOverriddenElements()` in `visualizeGroupColors` before we do the next round of visualization. I could add extra checks. I believe this logic is similar in the query building as well. I reverted this change because it seems that Ayush was trying to visualize different elements with the same color (red) and only one set was remaining. Replace within `overrideElements` is redundant for coloring groups because replacement only happens to elements that share the same color of what is being overridden. All groups have distinct colors, so nothing should be invoking replacement.

Reply:
As I understand you decided not to revert the change? I see that now `overrideElements` accepts optional `replace` with default value of `false`. Before it was `true`.

Reply:
Done